File: ciphers/caesar_cipher.go
package main

import (
	"fmt"
)

func Encrypt(text string, s int) string {
	result := ""
	for _, char := range text {
		if char >= 'A' && char <= 'Z' {
			char = rune((int(char) + s - 'A') % 26 + 'A')
			result += string(char)
		} else if char >= 'a' && char <= 'z' {
			char = rune((int(char) + s - 'a') % 26 + 'a')
			result += string(char)
		} else {
			result += string(char)
		}
	}
	return result
}

func Decrypt(text string, s int) string {
	return Encrypt(text, 26-(s%26))
}

func main() {
	text := "ATTACKATONCE"
	s := 4
	fmt.Println("Text :", text)
	fmt.Println("Shift:", s)
	cipher := Encrypt(text, s)
	fmt.Println("Cipher:", cipher)
	fmt.Println("Decrypted:", Decrypt(cipher, s))
}

File: ciphers/caesar_cipher.rs
fn encrypt(text: &str, s: u8) -> String {
    let mut result = String::new();
    for c in text.chars() {
        if c.is_ascii_uppercase() {
            let encoded = ((c as u8 - b'A' + s) % 26) + b'A';
            result.push(encoded as char);
        } else if c.is_ascii_lowercase() {
            let encoded = ((c as u8 - b'a' + s) % 26) + b'a';
            result.push(encoded as char);
        } else {
            result.push(c);
        }
    }
    result
}

fn decrypt(text: &str, s: u8) -> String {
    encrypt(text, (26 - (s % 26)) % 26)
}

fn main() {
    let text = "ATTACKATONCE";
    let s = 4;
    println!("Text : {}", text);
    println!("Shift: {}", s);
    let cipher = encrypt(text, s);
    println!("Cipher: {}", cipher);
    println!("Decrypted: {}", decrypt(&cipher, s));
}
